### PR TITLE
UPSTREAM: <drop>: remove the openshift authenticator from the apiserver

### PR DESCRIPTION
**Pull request by stlaz**
In 4.8, we moved the authenticator to be configured via
webhookTokenAuthenticators to an endpoint in the oauth-apiserver,
this should now be safe to remove.

/assign @deads2k 
/cc @s-urbaniak @slaskawi

### CODE REVIEW

File: pkg/kubeapiserver/authenticator/config.go [hunk 1]
```diff
@@ -174,13 +174,6 @@ func (config Config) New() (authenticator.Request, *spec.SecurityDefinitions, er
 		}
 
 		tokenAuthenticators = append(tokenAuthenticators, webhookTokenAuth)
-	} else {
-		// openshift gets injected here as a separate token authenticator because we also add a special group to our oauth authorized
-		// tokens that allows us to recognize human users differently than machine users.  The openAPI is always correct because we always
-		// configuration service account tokens, so we just have to create and add another authenticator.
-		// TODO make this a webhook authenticator and remove this patch.
-		// TODO - remove in 4.7, kept here not to disrupt authentication during 4.5->4.6 upgrade
-		tokenAuthenticators = AddOAuthServerAuthenticatorIfNeeded(tokenAuthenticators, config.APIAudiences)
```
Comment (s-urbaniak) [inline pkg/kubeapiserver/authenticator/config.go:183]: the diff doesn't show everything, but is the method `AddOAuthServerAuthenticatorIfNeeded` also removed?

Comment (s-urbaniak) [reply pkg/kubeapiserver/authenticator/config.go:183]: ah yes it is :+1:

**Review by slaskawi — APPROVED** [review]

File: pkg/kubeapiserver/authenticator/config.go (hunk 1 above)
Comment (stlaz) [reply pkg/kubeapiserver/authenticator/config.go:183]: yes, in `pkg/kubeapiserver/authenticator/patch_authenticator.go`

**Pull request merged**